experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+"""
+second_inclusion( G::Glueing ) = G.inclusionSecondPatch
+@Markdown.doc """
+    glueing_isomorphism( G::Glueing ) = G.glueingIsomorphism
+
+Returns the glueing isomorphism for the two realizations of the overlap U ∩ V 
+of the two patches. These two realizations come from considering U ∩ V as a 
+localization of both either U and of V.
+"""
+glueing_isomorphism( G::Glueing ) = G.glueingIsomorphism
+
+@Markdown.doc """
+    function Base.show( io::Base.IO, G::Glueing )
+
+Print the glueing G to io. If the field G.name is set, then the 
+print will be abbreviated to the output of this name.
+"""
+function Base.show( io::Base.IO, G::Glueing )
+  if isdefined( G, :name )
+    Base.print( io, G.name )
+    return
+  end
+  Base.println( io, "Glueing of" )
+  Base.println( io, G.firstPatch )
+  Base.println( io, "and" )
+  Base.println( io, G.secondPatch )
+  Base.println( io, "along" )
+  Base.println( io, domain(G.inclusionFirstPatch) )
+  return
+end
+
+function Base.copy( G::Glueing )
+  return Glueing( G.inclusionFirstPatch, G.inclusionSecondPatch, G.glueingIsomorphism )
+end
+
+################################################################################
+# Covering of a scheme.
+# It consists of a list of affine patches together with their glueings.
+# The glueings are stored in an array (which is usually half empty, since 
+# glueings need only be provided for ordered pairs of indices). 
+# The index methods for double arrays have been overloaded so that 
+# OrderedMultiindex can be used to address them.
+#
+@Markdown.doc """
+    mutable struct Covering
+
+Structure for storing a covering of a CoveredScheme X by affine 
+patches Uᵢ, i = 1,…,N. The necessary data consists of these affine 
+patches and the information on how to glue them pairwise via 
+glueins Gᵢⱼ: Uᵢ↩ Uᵢ∩ Uⱼ ↪ Uⱼ. 
+
+Once created, a covering must not be altered, since in general several 
+other objects, like e.g. vector bundles and morphisms of schemes, will 
+refer to such coverings and they need to rely on them to not change. 
+If one needs another covering derived from the given one, then a 
+new covering needs to be set up and stored. 
+"""
+mutable struct Covering
+  patches::Vector{AffineScheme}	# A list of open patches for this scheme
+  glueings::Matrix{Union{Glueing,Nothing}}     # A list of glueings between the patches 
+
+  # Variables for caching
+  name::String
+  						# listed above
+  function Covering( patches::Vector{AffineScheme}, glueings::Matrix{Union{Glueing,Nothing}} )
+    return new( patches, glueings )
+  end
+
+  function Covering()
+    C = new()
+    C.patches = AffineScheme[]
+    #C.glueings = Matrix{Union{Nothing,Glueing}}
+    return C
+  end
+
+  function Covering( X::AffineScheme ) 
+    C = new()
+    C.patches = [X]
+    C.glueings = Array{Nothing,2}(nothing,1,1)  # no glueing on the diagonal.
+    return C
+  end
+end
+
+function Base.copy( C::Covering )
+  # I'm afraid we need a copy here. Otherwise an alteration 
```

Comment:
Why are you "afraid"? It sounds as if this was something nasty, undesirable -- but isn't it rather completely expected? I only mention this because I feel there is a deeper concern you have in mind here?!